### PR TITLE
util-linux: update to 2.36.

### PR DESCRIPTION
**Pull request by ailiop-git**

### CODE REVIEW

File: srcpkgs/util-linux-libs/template [hunk 1]
```diff
@@ -13,8 +13,8 @@ short_desc="Miscellaneous linux utilities - meta libraries"
 maintainer="Enno Boland <gottox@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://www.kernel.org/pub/linux/utils/util-linux/"
-distfiles="${KERNEL_SITE}/utils/util-linux/v2.35/util-linux-${version}.tar.xz"
-checksum=21b7431e82f6bcd9441a01beeec3d57ed33ee948f8a5b41da577073c372eb58a
+distfiles="${KERNEL_SITE}/utils/util-linux/v${version}/util-linux-${version}.tar.xz"
```
Comment (sgn) [inline srcpkgs/util-linux-libs/template:16]: `v${version%.${version#*.*.}}`

**Comment by sgn** [comment]
`getopt` check failed on musl, `column` check failed on glibc.
I haven't looked further.

**Comment by ailiop-git** [comment]
> `getopt` check failed on musl, `column` check failed on glibc.

not sure what you mean "check failed", both binaries are compiling and running fine on both glibc/musl (for x86_64 at least).

**Comment by sgn** [comment]
On 2020-07-28 02:17:02-0700, ailiop-git <notifications@github.com> wrote:
> > `getopt` check failed on musl, `column` check failed on glibc.
> 
> not sure what you mean "check failed", both binaries are compiling
> and running fine on both glibc/musl (for x86_64 at least).

I meant:

	./xbps-src check util-linux
	./xbps-src -m musl check util-linux

-- 
Danh

**Comment by ailiop-git** [comment]
Ah ok, so the particular failure on glibc is on `column/invalid-multibyte` test, due to missing locale:

`/bin/sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)`

this was changed in commit `6ef9a9e18471b9 ("tests: (column) use actually invalid multibytes to test encoding")`, which means it has been failing since v2.35.

This can be fixed with:

```
sed -i s/^#C.UTF-8/C.UTF-8/ masterdir/etc/default/libc-locales
xbps-reconfigure -r masterdir/ glibc-locales -f
```

The second issue on musl, is the `getopt/options-posix_correctly `test failing. This is due to the ordering of the options processing, see diff of failing test below:

```
- -a -b '1' -c '2' --a-long --b-long '3' --c-long '' -- 'foo' 'bar' 'xyxxy' '--a-long'
+ -a -b '1' -c '2' --a-long --b-long '3' --c-long '' --a-long -- 'foo' 'bar' 'xyxxy'
```                                   
This happens as musl doesn't implement `POSIXLY_CORRECT` (which influences the ordering). It also means that the particular test has always been failing on musl.

**Comment by sgn** [comment]
On 2020-07-28 05:22:04-0700, ailiop-git <notifications@github.com> wrote:
> Ah ok, so the particular failure on glibc is on `column/invalid-multibyte` test, due to missing locale:
> 
> `/bin/sh: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)`
> 
> this was changed in commit `6ef9a9e18471b9 ("tests: (column) use
> actually invalid multibytes to test encoding")`, which means it has
> been failing since v2.35.
> 
> This can be fixed with:
> 
> ```
> sed -i s/^#C.UTF-8/C.UTF-8/ masterdir/etc/default/libc-locales
> xbps-reconfigure -r masterdir/ glibc-locales -f
> ```

OK, I'll ask for a change to glibc masterdir.

> 
> The second issue on musl, is the `getopt/options-posix_correctly
> `test failing. This is due to the ordering of the options
> processing, see diff of failing test below:
> 
> ```
> - -a -b '1' -c '2' --a-long --b-long '3' --c-long '' -- 'foo' 'bar' 'xyxxy' '--a-long'
> + -a -b '1' -c '2' --a-long --b-long '3' --c-long '' --a-long -- 'foo' 'bar' 'xyxxy'
> ```                                   
> This happens as musl doesn't implement `POSIXLY_CORRECT` (which
> influences the ordering). It also means that the particular test has
> always been failing on musl.

Hm, the check was added by me, and my daily driver is musl one,
I think it must be passed that time.


-- 
Danh

**Comment by ailiop-git** [comment]
> OK, I'll ask for a change to glibc masterdir.

Is this required? I assume we don't run checks by default in travis (and probably shouldn't).

> Hm, the check was added by me, and my daily driver is musl one, I think it must be passed that time.

I've just checked musl with util-linux 2.35.2 and getopt/options-posix_correctly fails as expected. There's also misc/mountpoint-default that fails there for whatever reason.

**Comment by ericonr** [comment]
> Is this required? I assume we don't run checks by default in travis (and probably shouldn't).

If it wouldn't lead to timeouts for most stuff, we should definitely run tests in travis. Still, we need to be able to run tests for essential software like this, at least locally, without having to figure out which tests are allowed to fail and whatnot.

**Comment by ailiop-git** [comment]
> If it wouldn't lead to timeouts for most stuff, we should definitely run tests in travis. Still, we need to be able to run tests for essential software like this, at least locally, without having to figure out which tests are allowed to fail and whatnot.

Sure, but this is more of an upstream development issue, rather than an issue at this packaging level. If those failures are a real concern (for example the getopt posix ordering doesn't seem worrysome), then they should be addressed upstream and properly fixed either by addressing the real root cause, or by marking them as known-fails under musl.

**Pull request closed without merging**